kasparovsyncd/sync/sync.go

```diff
@@ -412,21 +414,24 @@ func updateAddedChainBlocks(dbTx *dbaccess.TxContext, addedBlock *rpcmodel.Chain
 }
 
 func handleBlockAddedMsg(client *jsonrpc.Client, blockAdded *jsonrpc.BlockAddedMsg) error {
-	dbTx, err := dbaccess.NewTx()
-	if err != nil {
-		return err
-	}
-	defer dbTx.RollbackUnlessCommitted()
-
 	blockHash := blockAdded.Header.BlockHash()
-	blockExists, err := dbaccess.DoesBlockExist(dbaccess.NoTx(), blockHash.String())
+	blockExists, err := dbaccess.DoesBlockExist(database.NoTx(), blockHash.String())
 	if err != nil {
 		return err
 	}
 	if blockExists {
 		return nil
 	}
 
+	dbTx, err := database.NewTx()
+	if err != nil {
+		return err
+	}
+
+	defer func(dbTx *database.TxContext) {
```

Comment:
Why do you need the closure here?
You can simply do `defer dbTx.RollbackUnlessCommited`, no?

Reply:
I thought this is needed as we update dbtx, but since it's a pointer you're right. I'll fix that